controllers/idle_handlers.go

```diff
@@ -18,16 +18,57 @@ package controllers
 
 import (
 	"context"
+	"fmt"
+	"os"
 
 	lcav1alpha1 "github.com/openshift-kni/lifecycle-agent/api/v1alpha1"
+	"github.com/openshift-kni/lifecycle-agent/controllers/utils"
 	ctrl "sigs.k8s.io/controller-runtime"
 )
 
+// removeDeployment removes the deployment with the specified name
+// in case there is no deployment with the specified name returns false
+func (r *ImageBasedUpgradeReconciler) removeDeployment(osName string) (bool, error) {
+	status, err := r.RpmOSTreeClient.QueryStatus()
+	if err != nil {
+		return false, err
+	}
+	for index, deployment := range status.Deployments {
+		if deployment.OSName == osName {
+			ostreeUndeployCommand := fmt.Sprintf("ostree admin undeploy %d", index)
```

Comment:
Because the undeploy will shift the indexes, you'll need to either get a new status, or do the undeploy commands in reverse order